app/app.go

```diff
@@ -408,8 +429,59 @@ func NewOsmosisApp(
 		ante.DefaultSigVerificationGasConsumer,
 		encodingConfig.TxConfig.SignModeHandler(),
 		app.IBCKeeper,
+		BlockSDKAnteHandlerParams{
+			freeLane:      freeLane,
+			mevLane:       mevLane,
+			auctionKeeper: *app.AppKeepers.AuctionKeeper,
+			txConfig:      txConfig,
+		},
+	)
+
+	// update ante-handlers on lanes
+	opt := []base.LaneOption{
+		base.WithAnteHandler(anteHandler),
+	}
+	mevLane.WithOptions(opt...)
+	defaultLane.WithOptions(opt...)
+	freeLane.WithOptions(opt...)
+
+	// ABCI handlers
+	// prepare proposal
+	proposalHandler := blocksdkabci.NewProposalHandler(
+		app.Logger(),
+		txConfig.TxDecoder(),
+		txConfig.TxEncoder(),
+		lanedMempool,
+	)
+
+	// we use the block-sdk's PrepareProposal logic to build blocks
+	app.SetPrepareProposal(proposalHandler.PrepareProposalHandler())
+
+	// we use a no-op ProcessProposal, this way, we accept all proposals in avoidance
+	// of liveness failures due to Prepare / Process inconsistency. In other words,
+	// this ProcessProposal always returns ACCEPT.
+	app.SetProcessProposal(baseapp.NoOpProcessProposal())
```

Comment:
Note: this is set to no-op because the upstream priority nonce mempool that the block SDK utilizes under the hood does not natively support multi-fee denoms. As such the ordering may not be deterministic between proposal creation and verificaiton. This would require an integration with osmosis prices / slinky to turn on the ProcessProposalHandler.